### ISSUE TITLE
Pool Lighting Color Support

### ISSUE THREAD

**Issue by skonesam** [schemers/homebridge-screenlogic#5]
This is so great -- especially since the screen logic app is so incredibly, terribly, aggressively awful. Thanks for making it.

Think there's any way to get pool lighting colors into this? I have a workaround -- shortcut to run a script via ssh on an ancient Mac mini to the pool controller -- but that's exactly as clear and elegant as it sounds. 😉

**Comment by schemers** [comment]
thanks, glad you find it useful. I generally just turn the lights on/off and don't change them 😄 

 I took a quick look at the [library](https://github.com/parnic/node-screenlogic) I'm using to communicate with the controller, and it looks like it supports changing light color, so it should be possible.

Will probably need to explore the best way of archiving this. Could either by via a bunch mutually-exclusive switches (like, `Party`, `Romance`, etc) that let you select which type you want. Alternatively, there might be a way to configure the existing Pool/Spa light switches as a color-capable light. Will try and find some time to explore.

**Comment by ajcoolj** [comment]
@skonesam How do you SSH to the pool?

**Comment by skonesam** [comment]
It's not directly into the pool, it's via an old Mac mini I have set up in the basement as an always-on server. https://github.com/parnic/node-screenlogic

Not sure how much I played with the default example script… but I don't think it's much, since I don't really know how to code in javascript.

**Comment by schemers** [comment]
@skonesam which light commands would you expect to use? Just the ones that set colors?

The full light-related command set (which you’ve probably seen) looks like:

```
ScreenLogic.LIGHT_CMD_LIGHTS_OFF
ScreenLogic.LIGHT_CMD_LIGHTS_ON
ScreenLogic.LIGHT_CMD_COLOR_SET
ScreenLogic.LIGHT_CMD_COLOR_SYNC
ScreenLogic.LIGHT_CMD_COLOR_SWIM
ScreenLogic.LIGHT_CMD_COLOR_MODE_PARTY
ScreenLogic.LIGHT_CMD_COLOR_MODE_ROMANCE
ScreenLogic.LIGHT_CMD_COLOR_MODE_CARIBBEAN
ScreenLogic.LIGHT_CMD_COLOR_MODE_AMERICAN
ScreenLogic.LIGHT_CMD_COLOR_MODE_SUNSET
ScreenLogic.LIGHT_CMD_COLOR_MODE_ROYAL
ScreenLogic.LIGHT_CMD_COLOR_SET_SAVE
ScreenLogic.LIGHT_CMD_COLOR_SET_RECALL
ScreenLogic.LIGHT_CMD_COLOR_BLUE
ScreenLogic.LIGHT_CMD_COLOR_GREEN
ScreenLogic.LIGHT_CMD_COLOR_RED
ScreenLogic.LIGHT_CMD_COLOR_WHITE
ScreenLogic.LIGHT_CMD_COLOR_PURPLE
```

I’ve never used sync/swim/set-save/set-recall, so not sure it makes sense to include those.

I played around a little in the HomeKit simulator and it seems reasonable to create a grouping called “Light Colors” which groups them all in a single tile like so:

![IMG_2764](https://user-images.githubusercontent.com/249172/85960575-5f47f280-b959-11ea-97eb-095f478f9570.png)

Doing it that way prevents cluttering up a room with a bunch of new switches. You can also choose (within HomeKit) to show them as separate tiles if you really want to.  Could also provide a config option that lets you specify just the set of colors you wanted exposed(maybe as a comma-separated list like `Party,Romance,...`)

I’m still thinking those switches would just configure the desired light color, and not actually turn lights on/off.

**Comment by skonesam** [comment]
I think all of those sound good. When I made my little shortcut with menu, I definitely didn't include one or two in the list, so the comma-separated config (exactly like you have for unused Spa or other features) would be perfect.

A separate "switch" would be fine -- you're right that most of the time "on/off" is what is needed. (Also: it's still kinda magic to just say "turn on the pool lights" to your phone and have exactly that happen.)

**Comment by schemers** [comment]
Just released/published v2.1.0 which has support. You'll need to enable in the config. 

See [Note on light colors](https://github.com/schemers/homebridge-screenlogic#note-on-light-colors) in the README for details/limitations/etc.
